cli/src/main.rs

```diff
@@ -511,10 +537,222 @@ async fn run_test(test_args: TestArgs) -> anyhow::Result<i32> {
     Ok(exit_code)
 }
 
+async fn run_validate(validate_args: ValidateArgs) -> anyhow::Result<i32> {
```

Comment:
This file is getting unwieldy in length. Can we split out a file for this command and we'll split out the other commands into files later?

Also, I'd love for you to split out functions for each logical part of the command that you run so that it's easy to understand the flow

Reply:
Totally

Reply:
Alrighty, lmk if what I've got now is a bit easier on the eyes